Khan/wonder-blocks
# Allow user to specify what element should receieve focus after modal is closed

## Summary:
We need to find a way to pass the return the focus to a specific element outside
the modal component once this is closed.
The JIRA ticket mentions three possible routes - we're going with #2.

- Add a new prop to pass a reference to the element we want to return focus after the modal is closed (`closedFocusId`).

Issue: https://khanacademy.atlassian.net/browse/WB-1090

https://user-images.githubusercontent.com/13231763/153308190-0655802b-87ed-46ce-a4a6-7c183cc5404b.mov


## Test plan:
- `yarn jest packages/wonder-blocks-modal/src/components/__tests__/modal-launcher.test.js`

Storybook
- Go to http://localhost:6061/?path=/story/floating-modal-onepanedialog--with-closed-focus-id
- Open the dropdown
- Click on or press enter on "open modal"
- Click on or press enter on "close modal"
- Confirm that the "Focus here after close" button has focus
- Press enter
- Confirm that the `actions` menu now has focus
